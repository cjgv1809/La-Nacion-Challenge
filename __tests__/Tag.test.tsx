import { render, screen } from "@testing-library/react";
import TagComponent from "@/app/components/Tag";

const tag = {
  slug: "hello-world",
  text: "Hello, world",
  count: 1,
};

describe("Tag", () => {
  it("renders a tag", () => {
    render(<TagComponent tag={tag} />);

    const tagElement = screen.getByRole("link");
    expect(tagElement).toBeInTheDocument();
  });

  it("renders a different slug as part of the href", () => {
    render(<TagComponent tag={tag} />);

    const tagElement = screen.getByRole("link");
    expect(tagElement).not.toHaveAttribute("href", `/tags/${tag.slug}`);
  });

  it("renders the exact text passed to it", () => {
    render(<TagComponent tag={tag} />);

    const tagElement = screen.getByRole("link");
    expect(tagElement).toHaveTextContent(tag.text);
  });

  it("renders the exact count passed to it", () => {
    render(<TagComponent tag={tag} />);

    const tagElement = screen.getByRole("link");
    expect(tagElement).toHaveTextContent(tag.count.toString());
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<TagComponent tag={tag} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
