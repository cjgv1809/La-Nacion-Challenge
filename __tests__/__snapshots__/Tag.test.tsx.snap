// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag matches the snapshot 1`] = `
<DocumentFragment>
  <a
    href="/tema/hello-world"
  >
    Hello, world (1)
  </a>
</DocumentFragment>
`;
